api/app/Http/Controllers/CourseController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Course;
+use App\Models\CourseUser;
+use App\Models\Department;
+use Illuminate\Http\Request;
+
+class CourseController extends Controller
+{
+    public function index()
+    {
+        $courses = Course::all();
+
+        return response()->json($courses);
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate(
+            [
+                'course' => ['required'],
+                'department' => ['required'],
+                'user_ids' => ['required'],
+            ]
+        );
+        $course_id = Course::create(
+            [
+                'department_id' => Department::where('department', $request['department'])->get('id')->first()->id,
+                'course' => $request['course'],
+            ]
+        )->id;
+
+        $user_ids = array_filter(explode(',', $request['user_ids']));
+        foreach ($user_ids as $user_id) {
+            CourseUser::create(
+                [
+                    'user_id' => $user_id,
+                    'course_id' => $course_id,
+                ]
+            );
+        }
+
+        return response()->json(
+            [
+                'status' => 'Class created!',
+            ]
+        );
+    }
+
+    public function show($id)
+    {
+        $course = Course::findOrFail($id);
+        $course->courseUsers;
+
+        return response()->json(
+            [
+                'course_detail' => $course,
+
+            ]
+        );
+    }
+
+    public function update(Request $request, $id)
+    {
+        $request->validate(
+            [
+                'course' => ['required'],
+                'department' => ['required'],
+                'user_ids' => ['required'],
+            ]
+        );
+
+        $department_id = Department::where('department', $request['department'])->get('id')->first()->id;
```

Comment:
you can use `->first()` 
```suggestion
        $department_id = Department::where('department', $request['department'])->first()->pluck('id)
```